Backend/database/repositories/filesystem.go

```diff
@@ -3,14 +3,87 @@ package repositories
 
 import (
 	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
 	"strings"
+
+	"github.com/docker/docker/client"
 )
 
+const volume_path = "/var/lib/documents/data"
+
 // Implements IRepositoryInterface
 type FilesystemRepository struct {
 	embeddedContext
 }
 
+type DockerFileystemRepository struct {
+	cli *client.Client
```

Comment:
perhaps this should be changed to:
```go
dockerCli *client.Client
```
as its not immediately clear what this is for